statsmodels/discrete/count_model.py

```diff
@@ -0,0 +1,355 @@
+from __future__ import division
+
+__all__ = ["TruncatedPoisson"]
+
+import numpy as np
+import statsmodels.base.model as base
+import statsmodels.base.wrapper as wrap
+import statsmodels.regression.linear_model as lm
+from statsmodels.distributions import truncatedpoisson
+from statsmodels.discrete.discrete_model import (DiscreteModel, CountModel,
+                                                 Poisson, Logit, CountResults,
+                                                 L1CountResults, Probit,
+                                                 _discrete_results_docs)
+from statsmodels.tools.numdiff import (approx_fprime, approx_hess,
+                                       approx_hess_cs, approx_fprime_cs)
+
+class GenericTruncated(CountModel):
+    __doc__ = """
+    Generic Truncated model for count data
+
+    %(params)s
+    %(extra_params)s
+
+    Attributes
+    -----------
+    endog : array
+        A reference to the endogenous response variable
+    exog : array
+        A reference to the exogenous design.
+    """ % {'params' : base._model_params_doc,
+           'extra_params' :
+           """offset : array_like
+        Offset is added to the linear prediction with coefficient equal to 1.
+    exposure : array_like
+        Log(exposure) is added to the linear prediction with coefficient
+        equal to 1.
+
+    """ + base._missing_param_doc}
+
+    def __init__(self, endog, exog, truncation=0, offset=None,
```

Comment:
truncation params needs to be added to the docstring.
Specifically, it needs to specify whether the truncation point is in the support or out of support.
i.e. pmf(k) = 0 for k < truncation or for k<= truncation